packages/rich-text/src/plugins/CommandPalette/components/CommandList.tsx

```diff
@@ -0,0 +1,200 @@
+        event.preventDefault();
+        if (currIndex === buttons.length - 1) {
+          return;
+        }
+        setSelectedItem(buttons[currIndex + 1].id);
+        buttons[currIndex + 1].scrollIntoView({
+          block: 'nearest',
+          inline: 'start',
+        });
+      } else if (isHotkey('enter', event)) {
+        event.preventDefault();
+        if (currBtn) {
+          setSelectedItem('');
+          currBtn.click();
+        }
+      }
+      //TODO: handle shift+enter, which must be detected using separate events
+    }
+
+    if (commandItems.length) {
+      window.addEventListener('keydown', handleKeyDown);
+    }
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  }, [commandItems, container, selectedItem]);
+
+  return {
+    selectedItem,
+  };
+};
+
+export const CommandList = ({ query, editor }: CommandListProps) => {
+  const sdk = useSdkContext();
+  const container = React.useRef<HTMLDivElement>(null);
+  const commandItems = useCommands(sdk, query, editor);
+  const { selectedItem } = useCommandList(commandItems, container);
+
+  if (commandItems.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className={styles.container} tabIndex={-1} ref={container} contentEditable={false}>
+      {/*
+        We have to make it visually appear as if the buttons have focus, because we can not set both the
+        focus on the textarea and the focus on the button. In HTML you can only set focus on one element at a time.
+        So we have to manually tell the screenreader which item has received focus. The actual code of the
+        popover we will hide from the screenreader, since it is not accessibly relevant and to avoid reading
+        out everything twice.
+        We use role alert here because we want to make the screenreader immediately announce the selected
+        button, and also when the "fake focus" changes.
+       */}
+      <div role="alert">
+        <ScreenReaderOnly>
+          {/* TODO - show the label here and not the id */}
+          Richtext commands. Currently focused item: {selectedItem}. Press <kbd>enter</kbd> to
+          select, <kbd>arrows</kbd> to navigate, <kbd>escape</kbd> to close.
+        </ScreenReaderOnly>
+      </div>
+      <div aria-hidden={true}>
+        {/* eslint-disable-next-line jsx-a11y/no-autofocus -- we want to keep focus on text input*/}
+        <Popover isOpen={true} usePortal={false} autoFocus={false}>
+          {/* we need an empty trigger here for the positioning of the menu list */}
+          <Popover.Trigger>
+            <span />
+          </Popover.Trigger>
+          <Popover.Content className={styles.menuContent}>
+            <header className={styles.menuHeader}>
+              <SectionHeading marginBottom="none">Richtext commands</SectionHeading>
+            </header>
+            <div className={styles.menuList}>
+              {commandItems.map((item) => {
+                return 'group' in item ? (
```

Comment:
the nested ternary makes it hard to follow the rendering logic, can we use if-else here and give some semantic names to the conditions?